src/Subscription.hpp

```diff
@@ -27,16 +29,28 @@ namespace impl
  * CLASS DECLARATION
  **************************************************************************************/
 
-template <typename T, typename OnReceiveCb>
+template <typename T>
 class Subscription final : public SubscriptionBase
 {
+
 public:
-  Subscription(Node & node_hdl, CanardPortID const port_id, OnReceiveCb const & on_receive_cb)
+  using OnReceiveCallback_Default= std::function<void(T const &)>;
+  using OnReceiveCallback_Ext = std::function<void(T const &, TransferMetadata const &)>;
+  Subscription(Node & node_hdl, CanardPortID const port_id, OnReceiveCallback_Default const & on_receive_cb)
   : SubscriptionBase{CanardTransferKindMessage}
   , _node_hdl{node_hdl}
   , _port_id{port_id}
   , _on_receive_cb{on_receive_cb}
   { }
+
+  Subscription(Node & node_hdl, CanardPortID const port_id, OnReceiveCallback_Ext const & on_receive_cb_ext)
+  : SubscriptionBase{CanardTransferKindMessage}
+  , _node_hdl{node_hdl}
+  , _port_id{port_id}
+  , _on_receive_cb_ext{on_receive_cb_ext}
+  {
+    return_metadata = true;
```

Comment:
Don't really want to have such a boolean status variable. If metadata should be provided or not should be determined from the function signature of the receive callback alone.